nats.go

```diff
@@ -1999,10 +1999,7 @@ func (nc *Conn) respHandler(m *Msg) {
 // a wildcard. Caller is responsible for ensuring this is
 // only called once.
 func (nc *Conn) createRespMux() error {
-	nc.mu.Lock()
-	ginbox := nc.respSub
-	nc.mu.Unlock()
-	s, err := nc.Subscribe(ginbox, nc.respHandler)
+	s, err := nc.Subscribe(nc.respSub, nc.respHandler)
```

Comment:
Apologies, I now see that you were snapshotting the subject from the *Conn struct, which is probably needed since now line 2002 accesses it without a lock.